### PR TITLE
[wip][jit] Refactor math ops to use custom op bindings

### PR DESCRIPTION
**Pull request by driazati**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#19373 [jit] Refactor math ops to use custom op bindings**

Differential Revision: [D14987159](https://our.internmc.facebook.com/intern/diff/D14987159)

~~Now that we have a way to specify basic aliasing behavior for custom ops, we can register math ops as custom ops (and hopefully more things in register_prim_ops). Due to this we don't need to manually whitelist `aten` and `prim` for alias anaylsis, so the restriction when binding custom ops can be deleted.~~

TODO: right now aten ops are assumed to be pure and custom ops are assumed to be side effectful. we should merge these so everything is assumed to be side effectful unless stated otherwise

### CODE REVIEW

File: torch/csrc/jit/custom_operator.h [hunk 1]
```diff
@@ -182,26 +182,6 @@ Operator createOperator(
 
   auto schema = torch::jit::detail::inferAndCheckSchema<Traits>(schemaOrName);
 
-  // [custom operator aliasing] Currently, we have no way for the user to
```
Comment (eellison) [inline torch/csrc/jit/custom_operator.h]: Why did you delete this ?

Comment (driazati) [reply torch/csrc/jit/custom_operator.h]: #18589 makes it unnecessary

Comment (suo) [reply torch/csrc/jit/custom_operator.h]: I think we should do this clean up in a separate PR where you track down all references to this note and fix those up as well. Don't leave the pointers dangling!
